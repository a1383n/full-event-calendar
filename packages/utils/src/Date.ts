import { EventClass } from '@full-event-calendar/shared-ts'
import { areDatesInTheSameDate } from '.'
import { textChangeRangeIsUnchanged } from 'typescript'

export function floorDate(date: Date) {
  const newDate = new Date(date)
  newDate.setHours(0, 0, 0)
  return newDate
}

export function ceilDate(date: Date) {
  const newDate = new Date(date)
  newDate.setHours(23, 59, 59)
  return newDate
}

export function roundMinutesToMultipleOf5(date: Date) {
  const roundedMinutes = Math.round(date.getMinutes() / 5) * 5
  const newDate = new Date(date)
  newDate.setMinutes(roundedMinutes, 0, 0)
  return newDate
}

export function isDateToday(date: Date) {
  return areDatesInTheSameDate(date, new Date())
}

export function isEventRightOrLeftOrNone(event: EventClass, initialDate: Date) {
  /// in case the event is a dragging object
  if (!event.isAllDay()) return 'month-item-no-all-day'

  let flOWR = floorDate(initialDate)
  let Ceil = ceilDate(initialDate)
  let isEndOver = !areDatesInTheSameDate(event.end, initialDate)
  let isStartOver = !areDatesInTheSameDate(event.start, initialDate)

  if (event.start < flOWR && event.end > Ceil) {
    return 'month-both-arrow'
  } else if (isEndOver) {
    return 'month-right-arrow'
  } else if (isStartOver) {
    return 'month-left-arrow'
  }
  return 'month-no-arrow'
}

export function isEventRightOrLeftOrNoneRange(event: EventClass, start: Date, end: Date) {
  if (event.isAllDay) {
    if (!event.isAllDay()) return 'month-item-no-all-day'
  }

  let flOWR = floorDate(start)
  let Ceil = ceilDate(end)

  // let floweredEvent = floorDate(event.start)
  // let eventCeil = ceilDate(event.end)
  let floweredEvent = event.start
  let eventCeil = event.end
  if (event.start < flOWR && event.end > Ceil) {
    return 'month-both-arrow'
  } else if (floweredEvent >= flOWR && eventCeil > Ceil) {
    return 'month-right-arrow'
  } else if (floweredEvent <= flOWR && eventCeil < Ceil) {
    return 'month-left-arrow'
  }
  return 'month-no-arrow'
}

function addZeroPadd(num: number) {
  return num < 10 ? `0${num}` : num
}

export function getEventTimeRange(event: EventClass) {
  return `${addZeroPadd(event.start.getHours())}:${addZeroPadd(event.start.getMinutes())} - ${addZeroPadd(
    event.end.getHours()
  )}:${addZeroPadd(event.end.getMinutes())}`
}
