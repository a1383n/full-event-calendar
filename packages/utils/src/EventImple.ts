import { EventClass, SourceEvent } from '@full-event-calendar/shared-ts'
import { convertTZ } from './TimeZone'

export class EventImpl implements EventClass {
  start: Date
  end: Date
  name: string
  id: any
  duration: number // duration is in minuts
  sourceEvent: SourceEvent

  constructor(eventData: SourceEvent) {
    this.start = eventData.start
    this.end = eventData.end
    this.name = eventData.name
    this.id = eventData.id
    this.sourceEvent = eventData
    this.duration = Math.round((eventData.end.getTime() - eventData.start.getTime()) / 60000)
  }

  getEventLength(): any {
    return ''
  }

  isAllDay(): boolean {
    return this.duration / 60 >= 24
  }

  doesEventStartOn(date: Date) {
    return (
      this.start.getFullYear() === date.getFullYear() &&
      this.start.getMonth() === date.getMonth() &&
      this.start.getDate() === date.getDate()
    )
  }

  calculateHeight(calcFromZero: boolean = false) {
    let heightInPersentage
    if (calcFromZero) {
      heightInPersentage = ((this.end.getHours() * 60 + this.end.getMinutes()) / 60) * 100
    } else {
      heightInPersentage = (this.duration / 60) * 100
    }
    return `;height:  ${heightInPersentage}%;`
  }

  calculatePositionTop() {
    return `${this.getEventTopPositionIng()}`
  }

  checkOverLap(event: EventImpl) {
    const start1 = event.start
    const end1 = event.end
    const start2 = this.start
    const end2 = this.end
    return start1 < end2 && end1 > start2
  }

  convertDateByTimeZone(tz: string) {
    this.start = convertTZ(this.start, tz)
    this.end = convertTZ(this.end, tz)
  }

  private getEventTopPositionIng() {
    const eventColHeightInPersentage = (this.start.getHours() + this.start.getMinutes() / 60) * 100
    return `top: ${eventColHeightInPersentage}%`
  }
}

export type CalendarEvent = typeof EventImpl
