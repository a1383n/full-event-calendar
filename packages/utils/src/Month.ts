function getDaysOfMonth(date: Date, calendar: string) {
  const dateTimeFormat = new Intl.DateTimeFormat('en-US', {
    year: 'numeric',
    month: 'numeric',
    day: 'numeric',
    calendar
  })
  let parts = dateTimeFormat.formatToParts(date)
  return { ...extractYMD(parts), date: new Date(date) }
}

function extractYMD(parts: Intl.DateTimeFormatPart[]) {
  let year
  let month
  let day
  for (let i = 0; i < parts.length; i++) {
    //@ts-ignore
    if (parts[i].type === 'year' || parts[i].type === 'relatedYear') {
      year = parts[i].value
    } else if (parts[i].type === 'month') {
      month = parts[i].value
    } else if (parts[i].type === 'day') {
      day = parts[i].value
    }
  }

  return { year, month, day }
}

function extractMonthsDays(date: Date, calendar: string) {
  let selectedDate = date
  const convertedDate = getDaysOfMonth(selectedDate, calendar)
  const monthDays = [convertedDate]

  let Month = Number(convertedDate.month)

  const dataeCopy0 = new Date(selectedDate)
  for (let i = 40; i > 0; i--) {
    dataeCopy0.setDate(dataeCopy0.getDate() - 1)
    const convertedDate = getDaysOfMonth(dataeCopy0, calendar)
    const MonthNumber = Number(convertedDate.month)
    if (Month === MonthNumber) {
      monthDays.unshift(convertedDate)
    } else {
      break
    }
  }

  const dataeCopy = new Date(selectedDate)

  for (let i = 0; i < 40; i++) {
    dataeCopy.setDate(dataeCopy.getDate() + 1)
    const convertedDate = getDaysOfMonth(dataeCopy, calendar)
    const MonthNumber = Number(convertedDate.month)
    if (Month === MonthNumber) {
      monthDays.push(convertedDate)
    } else {
      break
    }
  }
  return monthDays
}

export function getCalendarMonthDays(date: Date, calendar: string) {
  const monthDays = extractMonthsDays(date, calendar)
  const WeekDay = monthDays[0].date.getDay()
  const dataeCopy = new Date(monthDays[0].date)

  for (let i = WeekDay; i > 0; i--) {
    dataeCopy.setDate(dataeCopy.getDate() - 1)
    const convertedDate = getDaysOfMonth(dataeCopy, calendar)
    monthDays.unshift(convertedDate)
  }
  const dataeCopy2 = new Date(monthDays[monthDays.length - 1].date)
  const weeek2 = dataeCopy2.getDay()
  for (let i = 0; i < 6 - weeek2; i++) {
    dataeCopy2.setDate(dataeCopy2.getDate() + 1)

    const convertedDate = getDaysOfMonth(dataeCopy2, calendar)
    monthDays.push(convertedDate)
  }

  return monthDays
}

export const ArraysplitIntoChunks = <T extends any[]>(arr: T, chunkSize: number) => {
  let res = []

  for (let i = 0; i < arr.length; i += chunkSize) {
    const chunk = arr.slice(i, i + chunkSize)
    // do whatever
    res.push(chunk)
  }
  return res
}
