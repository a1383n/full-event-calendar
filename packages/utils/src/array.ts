export const ArraySplitIntoChunks = <T extends any[]>(arr: T, chunkSize: number) => {
  let res = []

  for (let i = 0; i < arr.length; i += chunkSize) {
    const chunk = arr.slice(i, i + chunkSize)
    // do whatever
    res.push(chunk)
  }
  return res
}
