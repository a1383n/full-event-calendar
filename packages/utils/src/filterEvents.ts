import { EventClass } from '@full-event-calendar/shared-ts'

export function getEventForAdate(events: EventClass[], targetDate: Date) {
  const filteredERvents = events.filter((event) => {
    return (
      areDatesInTheSameDate(event.start, targetDate) ||
      areDatesInTheSameDate(event.end, targetDate) ||
      (event.start < targetDate && event.end > targetDate)
    )
  })
  return filteredERvents
}

export function areDatesInTheSameDate(date1: Date, date2: Date) {
  return (
    date1.getFullYear() === date2.getFullYear() &&
    date1.getMonth() === date2.getMonth() &&
    date1.getDate() === date2.getDate()
  )
}

interface Event {
  start: Date
  end: Date
  // Other event properties
}
interface EventsProp extends Event {
  [key: string]: any
}

export const filterEventsByDateRange = (events: EventsProp[], startDate: Date, endDate: Date) => {
  const flooredStartDate = new Date(startDate.setHours(0, 0, 0))
  const CeildStartDate = new Date(endDate.setHours(23, 59, 59))
  return events.filter((event) => {
    const condition =
      (event.start >= flooredStartDate && event.start <= CeildStartDate) ||
      (event.end >= flooredStartDate && event.end <= CeildStartDate) ||
      (event.end >= CeildStartDate && event.start <= flooredStartDate)
    return condition
  })
}

export function isDateIncludedInaRange(date: EventClass, rangeStart: Date, rangeEnd: Date) {
  return date.start < rangeEnd && date.end > rangeStart
}

export function daysDiffInRange(date1: Date, date2: Date) {
  const newF = new Date(new Date(date1).setHours(0, 0, 0))
  const secondF = new Date(new Date(date2).setHours(0, 0, 0))
  //@ts-ignore
  return Math.round((secondF - newF) / (1000 * 60 * 60 * 24))
}

// export  function clickOutside(el:any, accessor:any) {
//   const onClick = (e:any) => !el.contains(e.target) && accessor()?.();
//   document.body.addEventListener("click", onClick);

//   onCleanup(() => document.body.removeEventListener("click", onClick));
// }
