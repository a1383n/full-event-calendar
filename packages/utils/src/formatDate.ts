export function formatDDMMYYYY(date: Date, calendar: string = 'gregory') {
  return new Intl.DateTimeFormat('en-US', { month: '2-digit', day: '2-digit', year: 'numeric', calendar }).format(date)
}
export function formatDM(date: Date, calendar: string) {
  // return new Intl.DateTimeFormat('en-US', { month: '2-digit', day: '2-digit', year: 'numeric',calendar }).format(date)
  return new Intl.DateTimeFormat('en-US', { month: 'short', weekday: 'short', calendar }).format(date)
}

export function formatDD(date: Date, calendar: string) {
  // return new Intl.DateTimeFormat('en-US', { month: '2-digit', day: '2-digit', year: 'numeric',calendar }).format(date)
  return new Intl.DateTimeFormat('en-US', { day: '2-digit', calendar }).format(date)
}
