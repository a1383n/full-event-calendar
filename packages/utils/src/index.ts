export { rightOrLeftInDateInRange,rightOrLeftInDate } from './Month utils/rageArrows'
export { NewDraggingController } from './Drag/DraggingEvent'
export {  getEventsInDate, areDatesInTheSameDate, sortEventByStart  } from './filterEvents/filters'
export { EventModeFilter } from './filterEvents/EventFilter'
export { formatDDMMYYYY, formatDM, formatDD,formatRange,formatToShortTime } from './format/Date'
export { EventImpl } from './EventImple'
export { convertTZ } from './TimeZone'
export { getCalendarMonthDays, getMonthName, extractMonthDates } from './Month utils/Month'
export { ArraySplitIntoChunks } from './array'
export { formatWeekDays, formatDayNumber } from './format/Weekend'
export { isDateIncludedInaRange, daysDiffInRange, filterEventsByDateRange } from './filterEvents/range'
export { formatNumber } from './format/number'
export {
  floorDate,
  ceilDate,
  roundMinutesToMultipleOf5,
  isDateToday,
  getDateTimeRange,
  getEventTimeRange
} from './Date'
export { getWeekDates } from './week'
