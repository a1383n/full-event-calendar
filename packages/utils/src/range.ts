import { EventClass } from '@full-event-calendar/shared-ts'
import { ceilDate, floorDate } from '.'

interface Event {
  start: Date
  end: Date
  // Other event properties
}
interface EventsProp extends Event {
  [key: string]: any
}

export const filterEventsByDateRange = (events: EventsProp[], startDate: Date, endDate: Date) => {
  const flooredStartDate = floorDate(startDate)
  const CeildStartDate = ceilDate(endDate)
  return events.filter((event) => {
    const condition =
      (event.start >= flooredStartDate && event.start <= CeildStartDate) ||
      (event.end >= flooredStartDate && event.end <= CeildStartDate) ||
      (event.end >= CeildStartDate && event.start <= flooredStartDate)
    return condition
  })
}

export function isDateIncludedInaRange(date: EventClass, rangeStart: Date, rangeEnd: Date) {
  const flooredStart = floorDate(rangeStart)
  const ceilEnd = ceilDate(rangeEnd)
  return date.start < ceilEnd && date.end > flooredStart
}

export function daysDiffInRange(date1: Date, date2: Date) {
  const newF = floorDate(date1)
  const secondF = floorDate(date2)
  //@ts-ignore
  return Math.round((secondF - newF) / (1000 * 60 * 60 * 24))
}
