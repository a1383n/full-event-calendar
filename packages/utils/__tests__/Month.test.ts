import { describe, expect, test } from 'vitest'
import { getCalendarMonthDays } from '../src'

test('test month object', () => {
  const date1 = new Date(' Aug 01 2001 01:00:00')

  expect(getCalendarMonthDays(date1, 'persian')).toMatchSnapshot()
  expect(getCalendarMonthDays(date1)).toMatchSnapshot()
})
