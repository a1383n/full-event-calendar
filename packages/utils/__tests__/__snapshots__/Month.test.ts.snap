// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test month object 1`] = `
[
  {
    "date": 2001-07-21T20:30:00.000Z,
    "day": "31",
    "month": "4",
    "year": "1380",
  },
  {
    "date": 2001-07-22T20:30:00.000Z,
    "day": "1",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-07-23T20:30:00.000Z,
    "day": "2",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-07-24T20:30:00.000Z,
    "day": "3",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-07-25T20:30:00.000Z,
    "day": "4",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-07-26T20:30:00.000Z,
    "day": "5",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-07-27T20:30:00.000Z,
    "day": "6",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-07-28T20:30:00.000Z,
    "day": "7",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-07-29T20:30:00.000Z,
    "day": "8",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-07-30T20:30:00.000Z,
    "day": "9",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-07-31T20:30:00.000Z,
    "day": "10",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-01T20:30:00.000Z,
    "day": "11",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-02T20:30:00.000Z,
    "day": "12",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-03T20:30:00.000Z,
    "day": "13",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-04T20:30:00.000Z,
    "day": "14",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-05T20:30:00.000Z,
    "day": "15",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-06T20:30:00.000Z,
    "day": "16",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-07T20:30:00.000Z,
    "day": "17",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-08T20:30:00.000Z,
    "day": "18",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-09T20:30:00.000Z,
    "day": "19",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-10T20:30:00.000Z,
    "day": "20",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-11T20:30:00.000Z,
    "day": "21",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-12T20:30:00.000Z,
    "day": "22",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-13T20:30:00.000Z,
    "day": "23",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-14T20:30:00.000Z,
    "day": "24",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-15T20:30:00.000Z,
    "day": "25",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-16T20:30:00.000Z,
    "day": "26",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-17T20:30:00.000Z,
    "day": "27",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-18T20:30:00.000Z,
    "day": "28",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-19T20:30:00.000Z,
    "day": "29",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-20T20:30:00.000Z,
    "day": "30",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-21T20:30:00.000Z,
    "day": "31",
    "month": "5",
    "year": "1380",
  },
  {
    "date": 2001-08-22T20:30:00.000Z,
    "day": "1",
    "month": "6",
    "year": "1380",
  },
  {
    "date": 2001-08-23T20:30:00.000Z,
    "day": "2",
    "month": "6",
    "year": "1380",
  },
  {
    "date": 2001-08-24T20:30:00.000Z,
    "day": "3",
    "month": "6",
    "year": "1380",
  },
]
`;

exports[`test month object 2`] = `
[
  {
    "date": 2001-07-28T20:30:00.000Z,
    "day": "29",
    "month": "7",
    "year": "2001",
  },
  {
    "date": 2001-07-29T20:30:00.000Z,
    "day": "30",
    "month": "7",
    "year": "2001",
  },
  {
    "date": 2001-07-30T20:30:00.000Z,
    "day": "31",
    "month": "7",
    "year": "2001",
  },
  {
    "date": 2001-07-31T20:30:00.000Z,
    "day": "1",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-01T20:30:00.000Z,
    "day": "2",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-02T20:30:00.000Z,
    "day": "3",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-03T20:30:00.000Z,
    "day": "4",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-04T20:30:00.000Z,
    "day": "5",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-05T20:30:00.000Z,
    "day": "6",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-06T20:30:00.000Z,
    "day": "7",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-07T20:30:00.000Z,
    "day": "8",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-08T20:30:00.000Z,
    "day": "9",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-09T20:30:00.000Z,
    "day": "10",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-10T20:30:00.000Z,
    "day": "11",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-11T20:30:00.000Z,
    "day": "12",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-12T20:30:00.000Z,
    "day": "13",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-13T20:30:00.000Z,
    "day": "14",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-14T20:30:00.000Z,
    "day": "15",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-15T20:30:00.000Z,
    "day": "16",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-16T20:30:00.000Z,
    "day": "17",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-17T20:30:00.000Z,
    "day": "18",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-18T20:30:00.000Z,
    "day": "19",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-19T20:30:00.000Z,
    "day": "20",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-20T20:30:00.000Z,
    "day": "21",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-21T20:30:00.000Z,
    "day": "22",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-22T20:30:00.000Z,
    "day": "23",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-23T20:30:00.000Z,
    "day": "24",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-24T20:30:00.000Z,
    "day": "25",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-25T20:30:00.000Z,
    "day": "26",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-26T20:30:00.000Z,
    "day": "27",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-27T20:30:00.000Z,
    "day": "28",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-28T20:30:00.000Z,
    "day": "29",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-29T20:30:00.000Z,
    "day": "30",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-30T20:30:00.000Z,
    "day": "31",
    "month": "8",
    "year": "2001",
  },
  {
    "date": 2001-08-31T20:30:00.000Z,
    "day": "1",
    "month": "9",
    "year": "2001",
  },
]
`;
